ntbs-service/Pages/Logout.cshtml.cs

```diff
@@ -0,0 +1,27 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.Extensions.Options;
+
+namespace ntbs_service.Pages
+{
+  public class LogoutModel : PageModel
+    {
+    private readonly string ReturnUrl;
+
+    public LogoutModel(IOptionsMonitor<AdfsOptions> options)
+    {
+        // We just want to return to the homepage (which will trigger going to login again)
+        ReturnUrl = options.CurrentValue.Wtrealm;
+    }
+
+    public RedirectResult OnGet()
+        {
+            // Eras the cookie ...
+            HttpContext.SignOutAsync();
+
+            // ... and sign out of adfs
+            return Redirect($"https://fs.softwire.com/adfs/ls/?wa=wsignout1.0&wreply={ReturnUrl}");
```

Comment:
Might it be worth moving this base url as a constant somewhere into the authentication directory of the project? Not sure where the best place is but it seems weird buried in this view model

Reply:
Yeah, good shout - it should be part of the properties with the other adfs urls!